llvm/lib/Target/SBF/SBFISelLowering.cpp

```diff
@@ -601,9 +608,27 @@ SDValue SBFTargetLowering::LowerCall(TargetLowering::CallLoweringInfo &CLI,
       assert(VA.isMemLoc());
 
       EVT PtrVT = DAG.getTargetLoweringInfo().getPointerTy(DAG.getDataLayout());
-      SDValue Const = DAG.getConstant(SBFRegisterInfo::FrameLength - VA.getLocMemOffset(), CLI.DL, MVT::i64);
-      SDValue PtrOff = DAG.getNode(ISD::SUB, CLI.DL, PtrVT, FramePtr, Const);
-      Chain = DAG.getStore(Chain, CLI.DL, Arg, PtrOff, MachinePointerInfo());
+      SDValue DstAddr;
+      MachinePointerInfo DstInfo;
+      if (Subtarget->getHasDynamicFrames()) {
+        // When dynamic frames are enabled, the frame size is only calculated
+        // after lowering instructions, so we must place arguments at the start
+        // of the frame.
+        int64_t Offset =
+            -(int64_t)VA.getLocMemOffset() - (PtrVT.getFixedSizeInBits() / 8);
```

Comment:
Can you use `static_cast<int64_t>(VA.getLocMemOffset())` instead of a C-style cast?